.github/workflows/test.yml

```diff
@@ -27,26 +27,33 @@ jobs:
         shell: bash
       # Setup the environment for the tests
       - name: Install stack
-        run: ghcup install stack recommended
+        run: |
```

Comment:
this shouldn't be 4 separate steps... do this all in one step where HLS is installed as well